plugins/Live/Model.php

```diff
@@ -324,8 +328,12 @@ private function getWhereClauseAndBind($whereClause, $bindIdSites, $idSite, $per
 
         // SQL Filter with provided period
         if (!empty($period) && !empty($date)) {
-            $currentSite = $this->makeSite($idSite);
-            $currentTimezone = $currentSite->getTimezone();
+            if ($idSite === 'all') {
+                $currentTimezone = Request::processRequest('SitesManager.getDefaultTimezone');
```

Comment:
Will be only executed once and also only for `all` sites which is currently only used by GDPR feature. I'd rather not inject it as it would be always the same anyway. Executing API request be actually the way to go for me as the output can then be modified by basically various events etc if needed.